src/librustc_mir/interpret/memory.rs

```diff
@@ -142,10 +142,12 @@ fn const_eval_static<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>>(
         instance,
         promoted: None,
     };
-    tcx.const_eval(ty::ParamEnv::reveal_all().and(gid)).map_err(|err| {
+    // use the raw query here to break validation cycles. Later uses of the static will call the
+    // full query anyway
+    tcx.const_eval_raw(ty::ParamEnv::reveal_all().and(gid)).map_err(|_| {
         // no need to report anything, the const_eval call takes care of that for statics
         assert!(tcx.is_static(def_id).is_some());
-        EvalErrorKind::ReferencedConstant(err).into()
+        EvalErrorKind::ReferencedConstant.into()
     }).map(|val| {
         // FIXME We got our static (will be a ByRef), now we make a *copy*?!?
```

Comment:
The "copy" part btw is no longer true. `const_to_allocation` asserts that the constant is a `ByRef`, nowadays -- I think. Could you update the comment?

Reply:
I'll address this by nuking the entire query

Reply:
in a separate PR though